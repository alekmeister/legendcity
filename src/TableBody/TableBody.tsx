import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import { useAppSelector } from 'store/types';

export const TableBody: React.FC = () => {
  const data = useAppSelector((state) => state.users.users);
  return (
    <tbody>
      {data.map((user) => (
        <tr className="tbody_tr" key={uuidv4()}>
          {(Object.keys(user) as (keyof typeof user)[]).map((key) => (
            <td className="tbody_td" key={uuidv4()}>
              {user[key]}
            </td>
          ))}
        </tr>
      ))}
    </tbody>
  );
};
