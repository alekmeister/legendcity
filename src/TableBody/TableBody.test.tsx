// @ts-ignore
import { jest } from '@jest/globals';
import { TableBody } from 'TableBody/TableBody';
import { render } from '@testing-library/react';
import * as reduxHooks from 'react-redux';

const users = [
  {
    id: '1',
    firstName: 'Сергей',
    secondName: 'Валов',
    phone: '+79524216936',
    email: 'val321@mail.ru',
    password: '147852963.Qwe',
  },
  {
    id: '2',
    firstName: 'Иван',
    secondName: 'Архипов',
    phone: '+79528529647',
    email: 'ivan321q@mail.ru',
    password: '147852963.Qwe',
  },
];
jest.mock('react-redux');
const mokedUseSelector = jest.spyOn(reduxHooks, 'useSelector');
describe('TableBody', () => {
  it('should create TableBody with empty users', () => {
    mokedUseSelector.mockReturnValue([]);
    const component = render(<TableBody />);
    expect(component).toMatchSnapshot();
  });
  it('should create TableBody with users', () => {
    mokedUseSelector.mockReturnValue(users);
    const component = render(<TableBody />);
    expect(component).toMatchSnapshot();
  });
});
