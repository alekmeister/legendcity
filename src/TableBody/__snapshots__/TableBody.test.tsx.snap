// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody should create TableBody with empty users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tbody />
    </div>
  </body>,
  "container": <div>
    <tbody />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TableBody should create TableBody with users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tbody>
        <tr
          class="tbody_tr"
        >
          <td
            class="tbody_td"
          >
            1
          </td>
          <td
            class="tbody_td"
          >
            Сергей
          </td>
          <td
            class="tbody_td"
          >
            Валов
          </td>
          <td
            class="tbody_td"
          >
            +79524216936
          </td>
          <td
            class="tbody_td"
          >
            val321@mail.ru
          </td>
          <td
            class="tbody_td"
          >
            147852963.Qwe
          </td>
        </tr>
        <tr
          class="tbody_tr"
        >
          <td
            class="tbody_td"
          >
            2
          </td>
          <td
            class="tbody_td"
          >
            Иван
          </td>
          <td
            class="tbody_td"
          >
            Архипов
          </td>
          <td
            class="tbody_td"
          >
            +79528529647
          </td>
          <td
            class="tbody_td"
          >
            ivan321q@mail.ru
          </td>
          <td
            class="tbody_td"
          >
            147852963.Qwe
          </td>
        </tr>
      </tbody>
    </div>
  </body>,
  "container": <div>
    <tbody>
      <tr
        class="tbody_tr"
      >
        <td
          class="tbody_td"
        >
          1
        </td>
        <td
          class="tbody_td"
        >
          Сергей
        </td>
        <td
          class="tbody_td"
        >
          Валов
        </td>
        <td
          class="tbody_td"
        >
          +79524216936
        </td>
        <td
          class="tbody_td"
        >
          val321@mail.ru
        </td>
        <td
          class="tbody_td"
        >
          147852963.Qwe
        </td>
      </tr>
      <tr
        class="tbody_tr"
      >
        <td
          class="tbody_td"
        >
          2
        </td>
        <td
          class="tbody_td"
        >
          Иван
        </td>
        <td
          class="tbody_td"
        >
          Архипов
        </td>
        <td
          class="tbody_td"
        >
          +79528529647
        </td>
        <td
          class="tbody_td"
        >
          ivan321q@mail.ru
        </td>
        <td
          class="tbody_td"
        >
          147852963.Qwe
        </td>
      </tr>
    </tbody>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
